expressjs/db/migrations/20200526095904_rebuild.js

```diff
@@ -0,0 +1,113 @@
+    table.uuid('id').notNull().primary()
+    table.uuid('organization_id').notNull().references('organizations.id').onDelete('CASCADE');
+    table.string('info_website_url')
+    table.string('reference_website_url')
+    table.string('api_endpoint_url')
+    table.json('region_coordinates')
+    table.integer('notification_threshold_percent').defaultTo(66);
+    table.integer('notification_threshold_count').defaultTo(6);
+    table.integer('chunking_in_seconds').defaultTo(43200);
+    table.integer('days_to_retain_records').defaultTo(30);
+    table.timestamp('updated_at').defaultTo(knex.fn.now())
+    table.timestamp('created_at').defaultTo(knex.fn.now())
+  }).then(() => knex.raw(onUpdateTrigger('settings')));
+}
+
+function buildUsers(knex) {
+  return knex.schema.createTable('users', function (table) {
+    table.uuid('id').notNull().primary()
+    table.uuid('organization_id').notNull().references('organizations.id').onDelete('CASCADE');
+    table.string('username', 64)
+    table.string('email', 128)
+    table.string('password', 60)
+    table.string('maps_api_key')
+    table.boolean('is_admin')
+    table.timestamp('updated_at').defaultTo(knex.fn.now())
+    table.timestamp('created_at').defaultTo(knex.fn.now())
+  }).then(() => knex.raw(onUpdateTrigger('users')));
+  
+}
+
+function buildPublications(knex) {
+  return knex.schema.createTable('publications', function (table) {
+    table.increments('id').notNull().primary();
+    table.uuid('organization_id').notNull().references('organizations.id').onDelete('CASCADE');
+    table.timestamp('start_date').notNullable();
+    table.timestamp('end_date').notNullable();
+    table.timestamp('publish_date').notNullable();
+    table.timestamp('updated_at').defaultTo(knex.fn.now())
+    table.timestamp('created_at').defaultTo(knex.fn.now())
+  }).then(() => knex.raw(onUpdateTrigger('publications')));
+}
+
+function buildCases(knex) {
+  let createQuery = `
+    DO $$ BEGIN
+      CREATE TYPE state_type AS ENUM ('unpublished', 'staging', 'published');
+    EXCEPTION
+        WHEN duplicate_object THEN null;
+    END $$;
+    
+    CREATE TABLE cases(
+      id UUID PRIMARY KEY,
+      organization_id UUID NOT NULL REFERENCES organizations(id) ON DELETE CASCADE,
+      publication_id INT REFERENCES publications(id) ON DELETE CASCADE,
+      state state_type NOT NULL,
+      created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+      updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP
+    ) WITH (oids = false);
+  `;
+  return knex.raw(createQuery)
+    .then(() => knex.raw(onUpdateTrigger('cases')));
+}
+
+function buildTrails(knex) {
+  return knex.schema.createTable('trails', function (table) {
+    table.increments('id').notNull().primary();
+    table.uuid('case_id').notNull().references('cases.id').onDelete('CASCADE');
+    table.specificType('coordinates', 'geometry(point, 4326)');
+    table.timestamp('time');
+    table.string('hash')
+    table.timestamps(true, true);
```

Comment:
Should we change this to the same pattern?
```
    table.timestamp('updated_at').defaultTo(knex.fn.now());
    table.timestamp('created_at').defaultTo(knex.fn.now());
```

Reply:
Yes, good catch.